### PR TITLE
[FLINK-13920] Move release list into _config.yml 

### PR DESCRIPTION
**Pull request by zentol**
Moves the date for the [All stable releases](https://flink.apache.org/downloads.html#all-stable-releases) section into _config.yml, to reduce the number of files we have to updated manually when adding a new release.

### CODE REVIEW

File: downloads.md [hunk 1]
```diff
@@ -176,63 +176,31 @@ Note that the community is always open to discussing bugfix releases for even ol
...
+<ul>
+{% for flink_release in flink_releases %}
+<li>
+{% if flink_release.version_short %}
+Flink {{ flink_release.version_long }} - {{ flink_release.release_date }} 
+(<a href="https://archive.apache.org/dist/flink/flink-{{ flink_release.version_long }}/flink-{{ flink_release.version_long }}-src.tgz">Source</a>, 
+<a href="https://archive.apache.org/dist/flink/flink-{{ flink_release.version_long }}">Binaries</a>, 
+<a href="{{ site.DOCS_BASE_URL }}flink-docs-release-{{ flink_release.version_short }}">Docs</a>, 
+<a href="{{ site.DOCS_BASE_URL }}flink-docs-release-{{ flink_release.version_short }}/api/java">Javadocs</a>, 
+<a href="{{ site.DOCS_BASE_URL }}flink-docs-release-{{ flink_release.version_short }}/api/scala/index.html">ScalaDocs</a>)
```
[the remark this answers is not in the document]
Comment (zentol) [reply downloads.md]: `ScalaDocs` is the capitalization we currently use on the website :)

I can add it as a follow-up though.

**Comment by zentol** [comment]
Thank you for reminding me to update the release guide, I would've forgotten about it...

File: downloads.md [hunk 2]
```diff
@@ -176,63 +176,31 @@ Note that the community is always open to discussing bugfix releases for even ol
...
+{% else %}
+Flink {{ flink_release.version_long }} - {{ flink_release.release_date }} 
+(<a href="https://archive.apache.org/dist/flink/flink-{{ flink_release.version_long }}/flink-{{ flink_release.version_long }}-src.tgz">Source</a>, 
+<a href="https://archive.apache.org/dist/flink/flink-{{ flink_release.version_long }}">Binaries</a>)
```
[the remark this answers is not in the document]
Comment (zentol) [reply downloads.md]: true, but getting the formatting (specifically spaces) to work with that is kinda tricky.

The naive approach
```
(Flink release,
Source,
Binaries

if version_short then
  , Docs
  Javadocs
  Scaladocs
fi
)
```
results in `(Flink release, Source, Binaries , Docs, Javadocs, Scaladocs )`.

Note the extra space after `Binaries` and before `)`.

I couldn't find a quick fix for that; unless you know a way to do that I'd stick to the current version.

[the remark this answers is not in the document]
Comment (tillrohrmann) [reply downloads.md]: I think the following code does the trick:

```
{% for flink_release in flink_releases %}
<li>
Flink {{ flink_release.version_long }} - {{ flink_release.release_date }}
(<a href="https://archive.apache.org/dist/flink/flink-{{ flink_release.version_long }}/flink-{{ flink_release.version_long }}-src.tgz">Source</a>,
<a href="https://archive.apache.org/dist/flink/flink-{{ flink_release.version_long }}">Binaries</a>{% if flink_release.version_short %}, <a href="{{ site.DOCS_BASE_URL }}flink-docs-release-{{ flink_release.version_short }}">Docs</a>, <a href="{{ site.DOCS_BASE_URL }}flink-docs-release-{{ flink_release.version_short }}/api/java">Javadocs</a>, <a href="{{ site.DOCS_BASE_URL }}flink-docs-release-{{ flink_release.version_short }}/api/scala/index.html">Scaladocs</a>{% endif %})
</li>
{% endfor %}
```

The important bit is to put the different <a> tags not on different lines. I think with Jekyll >= 3.5 Liquid gives us control over whitespaces where this is no longer necessary.

[the remark this answers is not in the document]
Comment (zentol) [reply downloads.md]: I suppose that could work, but personally I'd prefer minor duplication over such a long line.

[the remark this answers is not in the document]
Comment (tillrohrmann) [reply downloads.md]: This is fine with me.

**Pull request merged**